build.sbt

```diff
@@ -17,8 +17,48 @@ lazy val settingsScalameter = Seq(
   connectInput in Benchmark := true,
   testFrameworks += new TestFramework("org.scalameter.ScalaMeterFramework"),
 )
+
+val scalaMeterVersion = "0.18"
+libraryDependencies ++= Seq(
+  "com.storm-enroute" %% "scalameter" % scalaMeterVersion % "bench",
+  "com.storm-enroute" %% "scalameter-core" % scalaMeterVersion % "bench"
+)
 // --- end - scalameter
 
+// CATS
+scalacOptions ++= Seq(
+  "-Ypartial-unification"
+)
+
+val catsVersion = "2.0.0-M1"
+libraryDependencies ++= Seq(
+  "org.typelevel" %% "cats-core" % catsVersion,
+  "org.typelevel" %% "cats-kernel" % catsVersion,
+  "org.typelevel" %% "cats-macros" % catsVersion,
+)
+// CATS -- end
+
+// Monocle
+//val monocleVersion = "1.5.0" // 1.5.0-cats based on cats 1.0.x
+//
+//libraryDependencies ++= Seq(
+//  "com.github.julien-truffaut" %%  "monocle-core"  % monocleVersion,
+//  "com.github.julien-truffaut" %%  "monocle-macro" % monocleVersion,
+//  "com.github.julien-truffaut" %%  "monocle-law"   % monocleVersion % "test"
+//)
+// Monocle --- end
+
+// Deep learning
+val dl4jVersion = "1.0.0-beta4"
+libraryDependencies ++= Seq(
+  "org.deeplearning4j" % "deeplearning4j-core" % dl4jVersion,
+  "org.nd4j" % "nd4j-native-platform" % dl4jVersion,
+  // @see https://deeplearning4j.org/docs/latest/deeplearning4j-config-cudnn
+  "org.nd4j" % "nd4j-cuda-10.1-platform" % dl4jVersion,
+  "org.deeplearning4j" % "deeplearning4j-cuda-10.0" % dl4jVersion
+)
+// Deep Learning -- -end
```

Comment:
too many vulnerabilities at the moment. 
Comment out and re-consider it in the next PR. 
at the moment just doing some pre-DL refactor.